Climate triggers: started (Labs feature) (#45421)
home-assistant/home-assistant.github.io: source/_triggers/climate.started_cooling.markdown

---
title: "Thermostat started cooling"
trigger: climate.started_cooling
domain: climate
description: "Triggers after one or more thermostats start cooling."
related_triggers:
  - climate.started_heating
  - climate.started_drying
  - climate.hvac_mode_changed
---

> The **Thermostat started cooling** trigger fires after a thermostat {% term entity %} begins actively cooling. This trigger monitors the `hvac_action` attribute rather than the HVAC mode—a thermostat can be set to **Cool** mode but still be idle if the current temperature already meets the target. The trigger only fires when the thermostat actually starts producing cool air.

Reviewer: ```suggestion
The **Thermostat started cooling** trigger fires after a thermostat {% term entity %} begins actively cooling. This trigger monitors the `hvac_action` attribute rather than the HVAC mode. A thermostat can be set to **Cool** mode but still be idle if the current temperature already meets the target. The trigger only fires when the thermostat actually starts producing cool air.
```

3. In the **When** section, select **Add trigger**.
4. Select what you want to monitor. Under **By target** (see [Targets](#targets)), pick the area your thermostat is in (like your bedroom or living room). You can also select a device, a specific entity, or a label.
5. From the triggers shown for that target, select **Thermostat started cooling**.
6. Under **Trigger when** (see [Behavior](#behavior-with-multiple-targets)), pick **Each**, **First**, or **All** to control how the trigger behaves when multiple thermostats are targeted.
7. Under **For at least**, set how long the thermostat must stay in the cooling state before the trigger fires. Leave it at zero to fire immediately.
8. Select **Save**.

### Options in the UI

{% options_ui %}
Trigger when:
  description: |
    When multiple thermostats are targeted, controls when the trigger fires:

    - **Each** (`any` in YAML, default): fires every time any targeted thermostat starts cooling.
    - **First** (`first` in YAML): fires only when the first of a group starts cooling.
    - **All** (`last` in YAML): fires only after every targeted thermostat is cooling.
For at least:

>   description: How long the thermostat must stay in the cooling state before the trigger fires. Default is `0` (fires immediately).

Reviewer: Actually the text input here is `0` hh, `00` mm and `00` ss:
```suggestion
  description: How long the thermostat must stay in the cooling state before the trigger fires. Default is zero (fires immediately).
```

{% endoptions_ui %}

{% include triggers/yaml_header.md %}

In YAML, **Thermostat started cooling** is referred to as `climate.started_cooling`. A basic example looks like this:

{% example %}
trigger: |
  trigger: climate.started_cooling
  target:
    entity_id: climate.living_room
{% endexample %}

This fires every time `climate.living_room` begins actively cooling.

### Options in YAML

YAML sometimes provides additional options for more complex use cases that are not available through the UI.

{% options_yaml %}
behavior:
  description: |
    When multiple thermostats are targeted, controls when the trigger fires:

    - `any` (**Each** in the UI, default): fire every time any targeted thermostat starts cooling.
    - `first` (**First** in the UI): fire only when the first thermostat starts cooling.

>     - `last` (**All** in the UI): fire only after every targeted thermostat is cooling.

Reviewer: ```suggestion
    - `any` (**Each** in the UI, default): fires every time any targeted thermostat starts cooling.
    - `first` (**First** in the UI): fires only when the first thermostat starts cooling.
    - `last` (**All** in the UI): fires only after every targeted thermostat is cooling.
```
